lib/analytics.js

```diff
@@ -498,12 +505,46 @@ Analytics.prototype.alias = function(to, from, options, fn) {
 
 /**
  * Register a `fn` to be fired when all the analytics services are ready.
+ * TODO: document this function for the new overload behaviour.
+ *
+ * @param {Function} fn
+ * @return {Analytics}
+ */
+
+Analytics.prototype.ready = function(a, b) {
+  if (is.fn(a)) {
```

Comment:
cc @segmentio/gateway what's the javascript convention here as to how to document this?

@tsholmes I think i've seen some code/PR by you that had this overloaded function behaviour.